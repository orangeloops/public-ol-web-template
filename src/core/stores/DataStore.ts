import * as _ from "lodash";
import {action, computed, observable, runInAction} from "mobx";
import moment from "moment";
import intl, {ReactIntlUniversalOptions} from "react-intl-universal";

import {GraphQLAPIClient} from "../apiclients/graphql/GraphQLAPIClient";
import {RefreshTokenRequest, RefreshTokenResponse} from "../apiclients/graphql/GraphQLAPIClient.types";
import {FetchUserRequest, FetchUserResponse, SignInRequest, SignInResponse} from "../apiclients/graphql/GraphQLAPIClient.types";
import {AppConfig} from "../AppConfig";
import {Locale} from "../locales/Locale";
import * as Models from "../models";
import {CoreHelper} from "../utils/CoreHelper";

export type AuthenticationState = {
  user?: Models.User;
  accessToken?: Models.AccessToken;
  loadingSignIn: boolean;
  loadingUser: boolean;
};

export class DataStore {
  private static instance: DataStore | undefined;

  @observable private initializing = false;
  @observable private initialized = false;

  @observable currentLocale: Locale = undefined as any;

  private authenticationInitialState: AuthenticationState = {
    loadingSignIn: false,
    loadingUser: false,
  };

  @observable authenticationState: AuthenticationState = _.cloneDeep(this.authenticationInitialState);

  // eslint-disable-next-line
  private constructor() {}

  static getInstance(): DataStore {
    if (!DataStore.instance) DataStore.instance = new DataStore();

    return DataStore.instance;
  }

  @computed
  get isInitializing(): boolean {
    return this.initializing;
  }

  @computed
  get isInitialized(): boolean {
    return this.initialized;
  }

  @computed
  get isAccessTokenExpired(): boolean {
    const {accessToken} = this.authenticationState;

    return !accessToken || moment().add(1, "hour").isSameOrAfter(accessToken.expiresAt, "minutes");
  }

  @action
  async initialize(): Promise<any> {
    this.initializing = true;

    await this.loadLocale();

    this.initializing = false;
    this.initialized = true;
  }

  @action
  private async loadLocale(locale?: Locale) {
    const {currentLocale} = this;

    const nextLocale = locale || AppConfig.Settings.Localization.defaultLocale;

    if (currentLocale && nextLocale && nextLocale.code === currentLocale.code) return;

    this.currentLocale = nextLocale;

    // https://github.com/moment/moment/issues/3624
    const hyphenIndex = nextLocale.code.indexOf("-");
    const momentLocale = moment.locale(hyphenIndex === -1 ? nextLocale.code : nextLocale.code.substr(0, hyphenIndex));

    const locales: Record<string, Locale> = {};

    for (const l of AppConfig.Settings.Localization.locales as Locale[]) locales[l.code] = l;

    const intlOptions: ReactIntlUniversalOptions = {
      currentLocale: nextLocale.code,
      locales,
      warningHandler: (warning: string) => {
        if (warning.indexOf("is not supported.") === -1) console.warn(warning);
      },
    };
    await intl.init(intlOptions);

    moment.updateLocale(momentLocale, {
      calendar: {
        lastDay: `[${CoreHelper.formatMessage("Common-yesterday")}]`,
        sameDay: `[${CoreHelper.formatMessage("Common-today")}]`,
        nextDay: `[${CoreHelper.formatMessage("Common-tomorrow")}]`,
        lastWeek: "L",
        nextWeek: "L",
        sameElse: "L",
      },
      relativeTime: {
        future: (text) => text,
        past: (text) => text,
        s: `%d ${CoreHelper.formatMessage("Common-second")}`,
        ss: `%d ${CoreHelper.formatMessage("Common-seconds")}`,
        m: `%d ${CoreHelper.formatMessage("Common-minute")}`,
        mm: `%d ${CoreHelper.formatMessage("Common-minutes")}`,
        h: `%d ${CoreHelper.formatMessage("Common-hour")}`,
        hh: `%d ${CoreHelper.formatMessage("Common-hours")}`,
        d: `%d ${CoreHelper.formatMessage("Common-day")}`,
        dd: (value) => {
          if (value < 7) {
            return `${value} ${CoreHelper.formatMessage("Common-days")}`;
          } else {
            const weeks = Math.round(value / 7);
            return `${weeks} ${weeks > 1 ? CoreHelper.formatMessage("Common-weeks") : CoreHelper.formatMessage("Common-week")}`;
          }
        },
        M: `%d ${CoreHelper.formatMessage("Common-month")}`,
        MM: `%d ${CoreHelper.formatMessage("Common-months")}`,
        y: `%d ${CoreHelper.formatMessage("Common-year")}`,
        yy: `%d ${CoreHelper.formatMessage("Common-years")}`,
      },
    });
  }

  @action
  async setLocale(locale: Locale) {
    const {currentLocale} = this;

    if (!currentLocale || currentLocale.code !== locale.code) await this.loadLocale(locale);
  }

  @action
  setAccessToken(accessToken: Models.AccessToken) {
    this.authenticationState.accessToken = accessToken;
  }

  @action
  reset() {
    this.authenticationState = _.cloneDeep(this.authenticationInitialState);
  }

  @action
  async signIn(request: SignInRequest): Promise<SignInResponse> {
    this.authenticationState.loadingSignIn = true;

    const response = await GraphQLAPIClient.signIn(request);

    runInAction(() => {
      this.authenticationState.loadingSignIn = false;

      if (response.success) this.authenticationState.accessToken = response.accessToken;
    });

    return response;
  }

  @action
  async fetchUser(request: FetchUserRequest): Promise<FetchUserResponse> {
    this.authenticationState.loadingUser = true;

    const response = await GraphQLAPIClient.fetchUser(request);

    runInAction(() => {
      this.authenticationState.loadingUser = false;

      if (response.success) this.authenticationState.user = response.user;
    });

    return response;
  }

  @action
  async refreshToken(request: RefreshTokenRequest): Promise<RefreshTokenResponse> {
    const response = await GraphQLAPIClient.refreshToken(request);

    runInAction(() => {
      if (response.success) this.setAccessToken(response.accessToken);
    });

    return response;
  }
}
