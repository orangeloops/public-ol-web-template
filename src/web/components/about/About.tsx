import * as React from "react";
import {observer} from "mobx-react";
import {WebHelper} from "../../utils/WebHelper";

import {Button, Modal} from "antd";

import styles from "./About.module.scss";

export type AboutProps = {
  onClose?: () => void;
};

export const About: React.FC<AboutProps> = observer((props) => {
  const {onClose} = props;

  const handleClose = React.useCallback(() => {
    if (onClose) onClose();
  }, [onClose]);

  const footer = (
    <Button type="primary" onClick={handleClose}>
      {WebHelper.formatMessage("Common-ok")}
    </Button>
  );

  return (
    <Modal title={WebHelper.formatMessage("About-title")} visible={true} width={300} footer={footer} onCancel={handleClose}>
      <div className={styles.container}>
        <div className={styles.companyContainer}>
          <div className={styles.text + " " + styles.bold}>{WebHelper.formatMessage("About-builtBy")}</div>
          <div className={styles.companyIcon} />
        </div>
      </div>
    </Modal>
  );
});
