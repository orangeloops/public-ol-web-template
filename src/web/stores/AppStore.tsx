import * as _ from "lodash";
import {action, observable} from "mobx";
import * as React from "react";
import {History, Location} from "history";
import {AppConfig} from "../AppConfig";
import {CoreHelper} from "../../core/utils/CoreHelper";
import {DataStore} from "../../core/stores/DataStore";

import {Icon, message} from "antd";
import {MessageType as AntMessageType} from "antd/lib/message";

// Internalization
import {en_US} from "../../core/locales/en_US";
import {es_ES} from "../../core/locales/es_ES";

import "moment/locale/en-ca";
import "moment/locale/es";

require("intl/locale-data/jsonp/en.js");
require("intl/locale-data/jsonp/es.js");

const antd_locales = {
  [en_US.code]: require("antd/lib/locale-provider/en_US"),
  [es_ES.code]: require("antd/lib/locale-provider/es_ES"),
};

export type MessageType = "success" | "error" | "info";

type MessageHide = {
  id: string;
  hide: AntMessageType;
};

export type AppMode = "mobile" | "desktop";

export type InputMode = "touch" | "mouse";

export type AppComponentType = "drawer_panel";

export type AppComponentState = {
  visible: boolean;
  props?: any;
};

export type AppStoreState = {
  mode: AppMode | undefined;
  inputMode: InputMode;

  messageHide: MessageHide[];
  stateByComponentType: Map<AppComponentType, AppComponentState>;
};

export class AppStore {
  private static instance: any = null;

  static history: History;
  static location: Location;

  static Intl = {
    antd_locales: antd_locales,
  };

  private initialConfig: any;

  mobileBreakpoint = 769; // sm breakpoint

  dataStore = new DataStore();

  // STATE
  @observable
  state: AppStoreState = {
    mode: this.evaluateMode(),
    inputMode: "touch",

    messageHide: [],
    stateByComponentType: new Map(),
  };

  private isTouch = false;
  private inputModeTimer: any | undefined;

  constructor() {
    if (!_.isNil(AppStore.instance)) return AppStore.instance;

    AppStore.instance = this;

    this.handlerWindowResize = this.handlerWindowResize.bind(this);
    this.handleTouchStart = this.handleTouchStart.bind(this);
    this.handleMouseOver = this.handleMouseOver.bind(this);

    window.addEventListener("resize", this.handlerWindowResize);

    if (document.documentElement) document.documentElement.setAttribute("data-browser", navigator.userAgent);

    document.addEventListener("touchstart", this.handleTouchStart);
    document.addEventListener("mouseover", this.handleMouseOver);

    this.initialConfig = _.cloneDeep(AppConfig);

    this.dataStore.setLocale(AppConfig.Settings.Localization.defaultLocale);

    return AppStore.instance;
  }

  private handleTouchStart() {
    const {inputMode} = this.state;

    clearTimeout(this.inputModeTimer);

    this.isTouch = true;
    if (inputMode !== "touch") this.setInputMode("touch");

    this.inputModeTimer = setTimeout(() => (this.isTouch = false), 500);
  }

  private handleMouseOver() {
    const {isTouch} = this;
    const {inputMode} = this.state;

    if (!isTouch && inputMode === "touch") this.setInputMode("mouse");
  }

  private evaluateMode(): AppMode {
    const {mobileBreakpoint} = this;

    return !_.isNil(mobileBreakpoint) ? (window.innerWidth < mobileBreakpoint ? "mobile" : "desktop") : "desktop";
  }

  private handlerWindowResize() {
    const mode = this.evaluateMode();

    this.setMode(mode);
  }

  navigateTo(path: string, replace = false) {
    console.assert(!_.isNil(path) && path.length > 0);

    const prefix = ``;

    const url = `${!path.startsWith(prefix) ? prefix : ""}${path.startsWith("/") ? "" : "/"}${path}`;

    if (replace) AppStore.history.replace(url);
    else AppStore.history.push(url);
  }

  // region ACTIONS

  @action
  setConfig(config: any) {
    console.assert(!_.isNil(config));

    CoreHelper.mergeWith(AppConfig, config, true, (value: any, sourceValue: any, key: string) => {
      switch (key) {
        case "schemas":
          return sourceValue;
        default:
          return _.isArray(value) ? sourceValue : undefined;
      }
    });

    this.dataStore.setLocale(AppConfig.Settings.Localization.defaultLocale);
  }

  @action
  resetConfig() {
    CoreHelper.mergeWith(AppConfig, this.initialConfig);
  }

  @action
  setInputMode(inputMode: InputMode) {
    this.state.inputMode = inputMode;
  }

  @action
  setMode(mode: AppMode) {
    console.assert(!_.isNil(mode));

    this.state.mode = mode;
  }

  @action
  showMessage(content: string | React.ReactNode, type: MessageType, duration: number = AppConfig.Modules.App.message.duration): string {
    if (this.state.messageHide.length >= 5) this.hideOldestMessage();

    const messageId = CoreHelper.getUUID();

    let fixedContent = content;
    if (typeof content === "string")
      fixedContent = (
        <div className="ol-message-wrapper">
          <span className="ol-message-content">{content}</span>
          {type === "error" && (
            <span className="ol-message-close-icon" onClick={() => this.hideMessage(messageId)}>
              <Icon type="close" />
            </span>
          )}
        </div>
      );

    let messageHide: MessageHide;

    switch (type) {
      case "success":
        messageHide = {
          id: messageId,
          hide: message.success(fixedContent, duration),
        };
        break;
      case "error":
        messageHide = {
          id: messageId,
          hide: message.error(fixedContent, 0),
        };
        break;
      case "info":
      default:
        messageHide = {
          id: messageId,
          hide: message.info(fixedContent, duration),
        };
        break;
    }

    this.state.messageHide.push(messageHide);
    return messageId;
  }

  @action
  hideMessage(id: string) {
    const messageToDeleteIndex = this.state.messageHide.findIndex(mh => mh.id === id);

    if (messageToDeleteIndex >= 0) {
      this.state.messageHide[messageToDeleteIndex].hide();

      this.state.messageHide.splice(messageToDeleteIndex, 1);
    }
  }

  @action
  hideOldestMessage() {
    this.state.messageHide[0].hide();

    this.state.messageHide.splice(0, 1);
  }

  @action
  hideAllMessages() {
    this.state.messageHide.forEach(mh => mh.hide());

    this.state.messageHide = [];
  }

  @action
  showComponent<P extends {} = {}>(type: AppComponentType, props?: P) {
    const {stateByComponentType} = this.state;

    const componentState = stateByComponentType.get(type);

    if (!_.isNil(componentState)) {
      componentState.visible = true;
      componentState.props = props;
    } else {
      stateByComponentType.set(type, {
        visible: true,
        props: props,
      });
    }
  }

  @action
  hideComponent(type: AppComponentType) {
    const {stateByComponentType} = this.state;

    const componentState = stateByComponentType.get(type);
    if (!_.isNil(componentState)) componentState.visible = false;
  }

  @action
  isComponentVisible(type: AppComponentType): boolean {
    const {stateByComponentType} = this.state;

    const component = stateByComponentType.get(type);

    return !_.isNil(component) && component.visible;
  }

  // endregion
}
