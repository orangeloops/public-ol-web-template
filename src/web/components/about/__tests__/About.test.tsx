import {mount} from "enzyme";
import * as React from "react";
import {About} from "../About";
import {DataStore} from "../../../../core/stores/DataStore";

describe("About", () => {
  test("Render correctly", async () => {
    await DataStore.getInstance().initialize();
    const wrapper = mount(<About />);

    expect(wrapper).toMatchSnapshot();
  });
});
