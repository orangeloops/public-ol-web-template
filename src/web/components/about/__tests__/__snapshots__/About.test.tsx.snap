// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Render correctly 1`] = `
<About>
  <Modal
    confirmLoading={false}
    footer={
      <Button
        block={false}
        ghost={false}
        htmlType="button"
        loading={false}
        onClick={[Function]}
        type="primary"
      >
        Ok
      </Button>
    }
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    title="About"
    transitionName="zoom"
    visible={true}
    width={300}
  >
    <Component
      closeIcon={
        <span
          className="ant-modal-close-x"
        >
          <Icon
            className="ant-modal-close-icon"
            type="close"
          />
        </span>
      }
      confirmLoading={false}
      footer={
        <Button
          block={false}
          ghost={false}
          htmlType="button"
          loading={false}
          onClick={[Function]}
          type="primary"
        >
          Ok
        </Button>
      }
      maskTransitionName="fade"
      okType="primary"
      onCancel={[Function]}
      onClose={[Function]}
      prefixCls="ant-modal"
      title="About"
      transitionName="zoom"
      visible={true}
      width={300}
      wrapClassName=""
    >
      <PortalWrapper
        visible={true}
      >
        <Portal
          getContainer={[Function]}
        >
          <Portal
            containerInfo={
              <div>
                <div>
                  <div
                    class="ant-modal-mask fade-appear"
                  />
                  <div
                    aria-labelledby="rcDialogTitle0"
                    class="ant-modal-wrap "
                    role="dialog"
                    tabindex="-1"
                  >
                    <div
                      class="ant-modal zoom-appear"
                      role="document"
                      style="width: 300px;"
                    >
                      <div
                        aria-hidden="true"
                        style="width: 0px; height: 0px; overflow: hidden;"
                        tabindex="0"
                      />
                      <div
                        class="ant-modal-content"
                      >
                        <button
                          aria-label="Close"
                          class="ant-modal-close"
                          type="button"
                        >
                          <span
                            class="ant-modal-close-x"
                          >
                            <i
                              aria-label="icon: close"
                              class="anticon anticon-close ant-modal-close-icon"
                            >
                              <svg
                                aria-hidden="true"
                                class=""
                                data-icon="close"
                                fill="currentColor"
                                focusable="false"
                                height="1em"
                                viewBox="64 64 896 896"
                                width="1em"
                              >
                                <path
                                  d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                                />
                              </svg>
                            </i>
                          </span>
                        </button>
                        <div
                          class="ant-modal-header"
                        >
                          <div
                            class="ant-modal-title"
                            id="rcDialogTitle0"
                          >
                            About
                          </div>
                        </div>
                        <div
                          class="ant-modal-body"
                        >
                          <div
                            class="container"
                          >
                            <div
                              class="companyContainer"
                            >
                              <div
                                class="text bold"
                              >
                                Phoenix is built by
                              </div>
                              <div
                                class="companyIcon"
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class="ant-modal-footer"
                        >
                          <button
                            class="ant-btn ant-btn-primary"
                            type="button"
                          >
                            <span>
                              Ok
                            </span>
                          </button>
                        </div>
                      </div>
                      <div
                        aria-hidden="true"
                        style="width: 0px; height: 0px; overflow: hidden;"
                        tabindex="0"
                      />
                    </div>
                  </div>
                </div>
              </div>
            }
          >
            <Dialog
              className=""
              closable={true}
              closeIcon={
                <span
                  className="ant-modal-close-x"
                >
                  <Icon
                    className="ant-modal-close-icon"
                    type="close"
                  />
                </span>
              }
              confirmLoading={false}
              destroyOnClose={false}
              footer={
                <Button
                  block={false}
                  ghost={false}
                  htmlType="button"
                  loading={false}
                  onClick={[Function]}
                  type="primary"
                >
                  Ok
                </Button>
              }
              getContainer={[Function]}
              getOpenCount={[Function]}
              keyboard={true}
              mask={true}
              maskClosable={true}
              maskTransitionName="fade"
              okType="primary"
              onCancel={[Function]}
              onClose={[Function]}
              prefixCls="ant-modal"
              title="About"
              transitionName="zoom"
              visible={true}
              width={300}
              wrapClassName=""
            >
              <div>
                <Animate
                  animation={Object {}}
                  component=""
                  componentProps={Object {}}
                  key="mask"
                  onAppear={[Function]}
                  onEnd={[Function]}
                  onEnter={[Function]}
                  onLeave={[Function]}
                  showProp="visible"
                  transitionAppear={true}
                  transitionEnter={true}
                  transitionLeave={true}
                  transitionName="fade"
                >
                  <AnimateChild
                    animation={Object {}}
                    key="mask"
                    transitionAppear={true}
                    transitionEnter={true}
                    transitionLeave={true}
                    transitionName="fade"
                  >
                    <LazyRenderBox
                      className="ant-modal-mask"
                      hiddenClassName="ant-modal-mask-hidden"
                      key="mask"
                      style={Object {}}
                      visible={true}
                    >
                      <div
                        className="ant-modal-mask"
                        style={Object {}}
                      />
                    </LazyRenderBox>
                  </AnimateChild>
                </Animate>
                <div
                  aria-labelledby="rcDialogTitle0"
                  className="ant-modal-wrap "
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  onMouseUp={[Function]}
                  role="dialog"
                  style={
                    Object {
                      "display": null,
                    }
                  }
                  tabIndex={-1}
                >
                  <Animate
                    animation={Object {}}
                    component=""
                    componentProps={Object {}}
                    key="dialog"
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="visible"
                    transitionAppear={true}
                    transitionEnter={true}
                    transitionLeave={true}
                    transitionName="zoom"
                  >
                    <AnimateChild
                      animation={Object {}}
                      key="dialog-element"
                      transitionAppear={true}
                      transitionEnter={true}
                      transitionLeave={true}
                      transitionName="zoom"
                    >
                      <LazyRenderBox
                        className="ant-modal "
                        key="dialog-element"
                        onMouseDown={[Function]}
                        role="document"
                        style={
                          Object {
                            "width": 300,
                          }
                        }
                        visible={true}
                      >
                        <div
                          className="ant-modal "
                          onMouseDown={[Function]}
                          role="document"
                          style={
                            Object {
                              "width": 300,
                            }
                          }
                        >
                          <div
                            aria-hidden="true"
                            style={
                              Object {
                                "height": 0,
                                "overflow": "hidden",
                                "width": 0,
                              }
                            }
                            tabIndex={0}
                          />
                          <div
                            className="ant-modal-content"
                          >
                            <button
                              aria-label="Close"
                              className="ant-modal-close"
                              onClick={[Function]}
                              type="button"
                            >
                              <span
                                className="ant-modal-close-x"
                              >
                                <Icon
                                  className="ant-modal-close-icon"
                                  type="close"
                                >
                                  <LocaleReceiver
                                    componentName="Icon"
                                  >
                                    <i
                                      aria-label="icon: close"
                                      className="anticon anticon-close ant-modal-close-icon"
                                    >
                                      <IconReact
                                        className=""
                                        type="close-o"
                                      >
                                        <svg
                                          aria-hidden="true"
                                          className=""
                                          data-icon="close"
                                          fill="currentColor"
                                          focusable="false"
                                          height="1em"
                                          key="svg-close"
                                          viewBox="64 64 896 896"
                                          width="1em"
                                        >
                                          <path
                                            d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                                            key="svg-close-svg-0"
                                          />
                                        </svg>
                                      </IconReact>
                                    </i>
                                  </LocaleReceiver>
                                </Icon>
                              </span>
                            </button>
                            <div
                              className="ant-modal-header"
                            >
                              <div
                                className="ant-modal-title"
                                id="rcDialogTitle0"
                              >
                                About
                              </div>
                            </div>
                            <div
                              className="ant-modal-body"
                            >
                              <div
                                className="container"
                              >
                                <div
                                  className="companyContainer"
                                >
                                  <div
                                    className="text bold"
                                  >
                                    Phoenix is built by
                                  </div>
                                  <div
                                    className="companyIcon"
                                  />
                                </div>
                              </div>
                            </div>
                            <div
                              className="ant-modal-footer"
                            >
                              <Button
                                block={false}
                                ghost={false}
                                htmlType="button"
                                loading={false}
                                onClick={[Function]}
                                type="primary"
                              >
                                <Wave>
                                  <button
                                    className="ant-btn ant-btn-primary"
                                    onClick={[Function]}
                                    type="button"
                                  >
                                    <span
                                      key=".0"
                                    >
                                      Ok
                                    </span>
                                  </button>
                                </Wave>
                              </Button>
                            </div>
                          </div>
                          <div
                            aria-hidden="true"
                            style={
                              Object {
                                "height": 0,
                                "overflow": "hidden",
                                "width": 0,
                              }
                            }
                            tabIndex={0}
                          />
                        </div>
                      </LazyRenderBox>
                    </AnimateChild>
                  </Animate>
                </div>
              </div>
            </Dialog>
          </Portal>
        </Portal>
      </PortalWrapper>
    </Component>
  </Modal>
</About>
`;
